### PR TITLE
Offer entity

### PR DESCRIPTION
**Pull request by JirkaVrbka**

### CODE REVIEW

File: src/main/java/cz/idomatojde/dao/OfferDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Override
+    public List<Offer> findByUser(User u) {
+        TypedQuery<Offer> query = em.createQuery(
+                "Select o from Offer o where o.owner = :userid",
+                Offer.class);
+
+        query.setParameter("userid", u);
+        return query.getResultList();
```
Comment (Michal-MK) [inline src/main/java/cz/idomatojde/dao/OfferDaoImpl.java]: Can be turned into a one-liner to improve readability, same goes for the other functions.

File: src/main/java/cz/idomatojde/entity/Offer.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
+package cz.idomatojde.entity;
+
+import com.sun.istack.NotNull;
```
Comment (Michal-MK) [inline src/main/java/cz/idomatojde/entity/Offer.java]: I would avoid non-standard imports.
@Column(nullable = false) or the javax.validation.constraints.NotNull seems better.

File: src/main/java/cz/idomatojde/entity/Offer.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+import javax.persistence.*;
```
Comment (Michal-MK) [inline src/main/java/cz/idomatojde/entity/Offer.java]: https://stackoverflow.com/questions/147454/why-is-using-a-wild-card-with-a-java-import-statement-bad

File: src/main/java/cz/idomatojde/entity/Offer.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+
+/*
+Created by Jiri Vrbka
+ */
+@Entity
+public class Offer {
+
+    @Id
+    @GeneratedValue(strategy= GenerationType.IDENTITY)
```
Comment (Michal-MK) [inline src/main/java/cz/idomatojde/entity/Offer.java]: Missing whitespace "strategy = GenerationType.IDENTITY"

File: src/main/java/cz/idomatojde/entity/Offer.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    @ManyToOne
+    private User owner;
+
+    @NotNull
+    private String title;
+
+    @NotNull
+    private String description;
+
+    private int price;
```
Comment (Michal-MK) [inline src/main/java/cz/idomatojde/entity/Offer.java]: Are we ok with integer prices?

File: src/main/java/cz/idomatojde/entity/Offer.java [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    private Date expirationDate;
+
+    @OneToMany
+    private TimetableEntry event;
+
+    private Category category;
+
+    private Integer capacity;
+
+    private Integer registered;
```
Comment (Michal-MK) [inline src/main/java/cz/idomatojde/entity/Offer.java:58]: What is the reasoning behing this being an "Integer" and price being "int"?

Comment (JirkaVrbka) [reply src/main/java/cz/idomatojde/entity/Offer.java:58]: Inspired from reference project

**Comment by Michal-MK** [comment]
I guess the code is now ok, merge it by squasing, so we do not introduce all the messy commits.

**Pull request merged**